instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java

```diff
@@ -0,0 +1,241 @@
+            return new StreamObserver<Helloworld.Response>() {
+              @Override
+              public void onNext(Helloworld.Response value) {
+                serverReceived.add(value.getMessage());
+
+                for (int i = 1; i <= serverMessageCount; i++) {
+                  observer.onNext(value);
+                }
+              }
+
+              @Override
+              public void onError(Throwable t) {
+                error.set(t);
+                observer.onError(t);
+              }
+
+              @Override
+              public void onCompleted() {
+                observer.onCompleted();
+                latch.countDown();
+              }
+            };
+          }
+        };
+
+    Server server = configureServer(ServerBuilder.forPort(0).addService(greeter)).build().start();
+    ManagedChannel channel = createChannel(server);
+    closer.add(() -> channel.shutdownNow().awaitTermination(10, TimeUnit.SECONDS));
+    closer.add(() -> server.shutdownNow().awaitTermination());
+
+    GreeterGrpc.GreeterStub client = GreeterGrpc.newStub(channel).withWaitForReady();
+
+    StreamObserver<Helloworld.Response> observer2 =
+        client.conversation(
+            new StreamObserver<Helloworld.Response>() {
+              @Override
+              public void onNext(Helloworld.Response value) {
+                clientReceived.add(value.getMessage());
+              }
+
+              @Override
+              public void onError(Throwable t) {
+                error.set(t);
+              }
+
+              @Override
+              public void onCompleted() {
+                latch.countDown();
+              }
+            });
+
+    for (int i = 1; i <= clientMessageCount; i++) {
+      Helloworld.Response message =
+          Helloworld.Response.newBuilder().setMessage("call " + i).build();
+      observer2.onNext(message);
+    }
+    observer2.onCompleted();
+
+    latch.await(10, TimeUnit.SECONDS);
+
+    assertThat(error).hasValue(null);
+    assertThat(serverReceived)
+        .containsExactlyElementsOf(
+            IntStream.rangeClosed(1, clientMessageCount)
+                .mapToObj(i -> "call " + i)
+                .collect(Collectors.toList()));
+    assertThat(clientReceived)
+        .containsExactlyElementsOf(
+            IntStream.rangeClosed(1, serverMessageCount)
+                .boxed()
+                .flatMap(
+                    unused ->
+                        IntStream.rangeClosed(1, clientMessageCount).mapToObj(i -> "call " + i))
+                .sorted()
+                .collect(Collectors.toList()));
+
+    List<Consumer<EventDataAssert>> events = new ArrayList<>();
+    for (int i = 1; i <= clientMessageCount * serverMessageCount + clientMessageCount; i++) {
+      long messageId = i;
+      events.add(
+          event ->
+              event
+                  .hasName("message")
+                  .hasAttributesSatisfying(
```

Comment:
do we have hasAttributesSatisfyingExactly on event yet?

Reply:
Not yet unfortunately noticed the gap with this PR